platform/android/detect.py

```diff
@@ -179,26 +178,22 @@ def mySpawn(sh, escape, cmd, args, env):
     env['RANLIB'] = tools_path + "/ranlib"
     env['AS'] = tools_path + "/as"
 
-    ndk_unified_headers = env['ndk_unified_headers']
-    ndk_version = get_ndk_version(env["ANDROID_NDK_ROOT"])
-
     common_opts = ['-fno-integrated-as', '-gcc-toolchain', gcc_toolchain_path]
 
-    if not ndk_unified_headers and ndk_version != None and ndk_version[0] >= 16:
-        ndk_unified_headers = True
-        print("Turning NDK unified headers on (starting from r16)")
-
     lib_sysroot = env["ANDROID_NDK_ROOT"] + "/platforms/" + env['ndk_platform'] + "/" + env['ARCH']
 
     ## Compile flags
 
-    if ndk_unified_headers:
+    ndk_version = get_ndk_version(env["ANDROID_NDK_ROOT"])
+    if ndk_version != None and LooseVersion(ndk_version) >= LooseVersion("15.0.4075724"):
```

Comment:
Well I merged too fast, this commit is incomplete. In the 2.1 one you have `from distutils.version import LooseVersion`, it's missing here.

Reply:
@akien-mga My bad, somehow the built didn't fail alerting of the missing import. [This PR](https://github.com/godotengine/godot/pull/12782) has it. Sorry about that.